lib/common/results.c

```diff
@@ -25,6 +25,47 @@
 G_DEFINE_QUARK(pcmk-rc-error-quark, pcmk__rc_error)
 G_DEFINE_QUARK(pcmk-exitc-error-quark, pcmk__exitc_error)
 
+// General (all result code types)
+
+/*!
+ * \brief Get the name and description of a given result code
+ *
+ * A result code can be interpreted as a member of any one of several families.
+ *
+ * \param[in]  code  The result code to look up
+ * \param[in]  type  How \p code should be interpreted
+ * \param[out] name  Where to store the result code's name
+ * \param[out] desc  Where to store the result code's description
+ *
+ * \return Standard Pacemaker return code
+ */
+int
+pcmk_result_get_strings(int code, enum pcmk_result_type type, const char **name,
+                        const char **desc)
+{
+    CRM_ASSERT((name != NULL) && (desc != NULL));
```

Comment:
For flexibility, we should allow name and desc to be NULL, and set them only if not (in case a caller needs just one of them)